src/bitcoind.cpp

```diff
@@ -150,9 +159,20 @@ bool AppInit(int argc, char* argv[])
 
             pid_t sid = setsid();
             if (sid < 0)
-                fprintf(stderr, "Error: setsid() returned %d errno %d\n", sid, errno);
+                fprintf(stderr, "Error: setsid() failed: %s\n", strerror(errno));
+
+            (void)chdir("/");
```

Comment:
I think the first option for this PR makes sense. That keeps this a refactor, not a change of behavior (which would break compatibility for users). In some later change the chdir could be re-added after implementing one of the other options.